### PR TITLE
Add an `amalgamate.sh` script to create an amalgamated `lighttpd` and transpile it

### PR DESCRIPTION
**Pull request by kkysen**
This adds a script, `amalgamate.sh`, to create an amalgamated `lighttpd` and transpile it.  We want to amalgamate it so that we only have one compilation unit and thus one transpiled Rust module so that we don't have any cross-module imports through `extern`s (vs. the standard `use`).  This way, static analysis will be able to see all function definitions from other modules and we can test static analysis out on much more of `lighttpd`.  This isn't meant as a permanent solution for how static analysis works—eventually, we'll probably want a tool that does the cross-module linking from `extern`s to `use`s and is more careful about everything—but for now, this mostly works amalgamation solution is the simplest for `lighttpd`.

The way this works is it

1. It first generates a `compile_commands.json` specifically for building the `lighttpd` binary.
2. Then for each `file` in that, it creates an amalgamated `lighttpd.amalgamated.c` that `#include`s all of those source `*.c` files.
  i. `lemon.c` is skipped, as this is a separate binary (a parser generator), so it has its own `main` that conflicts.
  ii.  For some reason, if `algo_xxhash.c` is `#include`d where it normally is, it doesn't define `XXH32`, but it does (as normal) when it's `#include`d first.
3. `lighttpd.amalgamated.c` is compiled to `lighttpd.amalgamated.exe`.
  i. `static const char hex_chars_uc[] = "0123456789ABCDEF"` is repeated verbatim in multiple source files, so I added `#ifndef` guards around them all (1a21e8833f722f65cfe48ac1cd0bf58e8f637b0c).
4. An `amalgamated.compile_commands.json` is created that only compiles `lighttpd.amalgamated.c`.  The flags are determined by unioning all of the flags used for all of the files.
5. Finally, `c2rust transpile` is used on `amalgamated.compile_commands.json` to transpile into `lighttpd_rust_amalgamated`.
  i. I didn't include the transpiled Rust in this PR yet.  I'll put it in a subsequent one, and it can be created by running `./amalgamate.sh`.

`./amalgamate.sh` uses `./amalgamate.mjs`, a Node JS script, which uses a modern `node` (tested on `v19.7.0`).  Since this should just be a one-off script to generate the transpiled and amalgamated `lighttpd`, I think it should be fine.

### CODE REVIEW

File: src/buffer.c [hunk 1]
```diff
@@ -7,7 +7,11 @@
 #include "sys-time.h"   /* strftime() */
 
 static const char hex_chars_lc[] = "0123456789abcdef";
+
+#ifndef HEX_CHARS_UC
```
Comment (oinoom) [inline src/buffer.c:11]: why are these necessary?

**Comment by oinoom** [comment]
i think it's worth adding the amalgamated files and transpiled files themselves too

File: src/buffer.c (hunk 1 above)
Comment (kkysen) [reply src/buffer.c:11]: > `static const char hex_chars_uc[] = "0123456789ABCDEF"` is repeated verbatim in multiple source files, so I added `#ifndef` guards around them all (1a21e8833f722f65cfe48ac1cd0bf58e8f637b0c).

Without the guards, `hex_chars_uc` is defined multiple times and so the compiler complains.  In a normal build, the `static` prevents them from clashing, but if it's in a single file, that doesn't work anymore.

**Comment by oinoom** [comment]
woops i see #3 does this

**Comment by kkysen** [comment]
> i think it's worth adding the amalgamated files and transpiled files themselves too

I put it in #3 instead since it can be added without this PR, and it might make reviewing this easier without the tons of changes from the other PR.

File: src/buffer.c (hunk 1 above)
Comment (oinoom) [reply src/buffer.c:11]: got it, thanks

**Review by oinoom — APPROVED** [review]
